redbot/core/settings_caches.py

```diff
@@ -59,23 +59,31 @@ async def set_prefixes(
 class I18nManager:
     def __init__(self, config: Config):
         self._config: Config = config
-        self._guild_locale: Dict[int, str] = {}
-        self._guild_regional_format: Dict[int, str] = {}
+        self._guild_locale: Dict[int, Union[str, None]] = {}
+        self._guild_regional_format: Dict[int, Union[str, None]] = {}
 
     async def get_locale(self, guild: Union[discord.Guild, None]) -> str:
         """Get the guild locale from the cache"""
-        if guild is None:
+        # Ensure global locale is in the cache
+        if None not in self._guild_locale:
+            global_locale = await self._config.locale()
+            self._guild_locale[None] = global_locale
+
+        if guild is None:  # Not a guild so cannot support guild locale
             # Return the bot's globally set locale if its None on a guild scope.
-            out = await self._config.locale()
-        elif guild.id in self._guild_locale:
-            out = self._guild_locale[guild.id]
-        else:
-            out = await self._config.guild(guild).locale()
+            return self._guild_locale[None]
+        elif guild.id in self._guild_locale:  # Cached guild
+            if self._guild_locale[guild.id] is None:
+                return self._guild_locale[None]
+            else:
+                return self._guild_locale[guild.id]
+        else:  # Uncached guild
+            out = await self._config.guild(guild).locale()  # No locale set
             if out is None:
-                out = await self._config.locale()
+                self._guild_locale[guild.id] = None
+                return self._guild_locale[None]
             else:
-                self._guild_locale[guild.id] = out
-        return out
+                return self._guild_locale[guild.id]
```

Comment:
Looks like you didn't update this method and only updated `get_regional_formatting()` method:
```suggestion
                self._guild_locale[guild.id] = out
                return out
```